### PR TITLE
[DOC] Mention in style guide the meaning of trailing underscore for attributes

### LINKED ISSUE #4291: Mention in style guide the meaning of trailing underscore for attributes
pointing to the scikit learn doc: https://scikit-learn.org/stable/developers/develop.html#estimated-attributes

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/pull/3761#discussion_r1505550097_

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4291

### CODE REVIEW

File: doc/Makefile [hunk 1]
```diff
@@ -68,7 +69,7 @@ sym_links:
 	# subdirectories of ../examples/ and creates a sym-link to
 	# ../examples/nilearn_cache/
 	-find ../examples/ -type d -mindepth 1 -maxdepth 1 -not -path "../examples/nilearn_cache" -exec ln -sf ../nilearn_cache "{}/nilearn_cache" \;
-	-mkdir -p ../examples/nilearn_cache/joblib
+	-mkdir -p ../examples/nilearn_cache/joblibvolunteers from the maintainers group who would like
```
Comment (Remi-Gau) [inline doc/Makefile:72]:
```suggestion
	-mkdir -p ../examples/nilearn_cache/joblib
```

File: doc/Makefile [hunk 2]
```diff
@@ -59,6 +59,7 @@ glm_reports:
 
 sym_links_datasets:
 	# copy datasets RST from nilearn package to doc folder
+	rm -fr modules/description
```
Comment (man-shu) [inline doc/Makefile:62]: Is this relevant here? Looks like a "spill" from other make file-related PRs?

Comment (Remi-Gau) [reply doc/Makefile:62]: ha yeah I was getting some troubles generating the doc locally

will remove for now and readd in a separate PR if I can confirm the problem

Comment (Remi-Gau) [inline doc/Makefile:62]:
```suggestion
```

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -389,6 +392,21 @@ This is also useful for writing unit tests.
...
+APIs of nilearn objects
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Estimated Attributes
+""""""""""""""""""""
+
+Attributes that have been estimated from the data
+must always have a name ending with trailing underscore.
+For example the coefficients of some regression estimator
+would be stored in a ``coef_`` attribute after ``fit`` has been called.
```
Comment (Remi-Gau) [inline CONTRIBUTING.rst:404]: Note that I reused the phrasing from scikit learn but I have not checked if it IS actually the case that all estimated attributes in our code base end with an underscore.

So I will turn the "must" in the sentence above to a "should" until we got that checked.

Could then open an issue to make sure we don't forget to check (and fix) any departure from this rule.

**Review by man-shu — APPROVED** [review]
Ok for me!

**Pull request merged**